docs/jupyter_execute/examples/Codegen/ref.ipynb

```diff
@@ -0,0 +1,282 @@
+       " 'test': \"\\n\\nMETADATA = {\\n    'author': 'jt',\\n    'dataset': 'test'\\n}\\n\\n\\ndef check(candidate):\\n    assert candidate([1.0, 2.0, 3.9, 4.0, 5.0, 2.2], 0.3) == True\\n    assert candidate([1.0, 2.0, 3.9, 4.0, 5.0, 2.2], 0.05) == False\\n    assert candidate([1.0, 2.0, 5.9, 4.0, 5.0], 0.95) == True\\n    assert candidate([1.0, 2.0, 5.9, 4.0, 5.0], 0.8) == False\\n    assert candidate([1.0, 2.0, 3.0, 4.0, 5.0, 2.0], 0.1) == True\\n    assert candidate([1.1, 2.2, 3.1, 4.1, 5.1], 1.0) == True\\n    assert candidate([1.1, 2.2, 3.1, 4.1, 5.1], 0.5) == False\\n\\n\"}"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from humaneval.humaneval import HumanEvalDataset\n",
+    "raw_dataset = HumanEvalDataset()\n",
+    "\n",
+    "problem = raw_dataset.data[0]\n",
+    "problem"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our workflow takes as input this `problem` dictionary and generates `finalized_code`.\n",
+    "\n",
+    "The evaluator function expects both `problem` and the `finalized_code`.\n",
+    "\n",
+    "> **Note:**\n",
+    ">\n",
+    "> Cognify will also forward workflow input to the evalautor function (if required in the function signature):\n",
+    "> - to cater for cases like *llm as a judge* where the question is also needed in the evaluation\n",
+    "\n",
+    "Thus we only need to pass `problem` as input and set ground truth to empty."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from humaneval.humaneval import HumanEvalDataset\n",
+    "import random\n",
+    "\n",
+    "@cognify.register_data_loader\n",
+    "def load_data():\n",
+    "    raw_dataset = HumanEvalDataset()\n",
+    "    size = len(raw_dataset.data)\n",
+    "    # shuffle the data\n",
+    "    random.seed(42)\n",
+    "    random.shuffle(raw_dataset.data)\n",
+    "    \n",
+    "    data = []\n",
+    "    for i in range(size):\n",
+    "        problem = raw_dataset.data[i]\n",
+    "        input = {'problem': problem}\n",
+    "        ground_truth = {}\n",
+    "        data.append((input, ground_truth))\n",
+    "    train, val, test = data[:40], data[40:60], data[60:]\n",
+    "    return train, val, test"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3.3 Config the optimizer\n",
+    "\n",
+    "Let's use the predefined search space for code generation, the search space includes:\n",
+    "\n",
+    "- Top Layer:\n",
+    "    - whether to spawn mutliple workers for each agent\n",
```

Comment:
```suggestion
    "    - whether to spawn multiple workers for each agent\n",
```

Maybe say "parallel sampling" instead of "spawn multiple workers"? I feel like parallel sampling is a more familiar term